### PR TITLE
docs: maximize signal-to-noise

### PR DESCRIPTION
**Pull request by sergey-shandar**
## Summary

Add signal-to-noise as a repository-wide design principle alongside simplicity and API quality.

- make high-level structure obvious
- put details and edge cases at the leaves rather than in the main flow
- prefer progressive understanding: abstraction first, structure second, details last

## Verification

Documentation-only change; no runtime tests required.

Changelog: none

### CODE REVIEW

File: DESIGN.md [hunk 1]
```diff
@@ -1,20 +1,21 @@
...
 1. [Simplicity first](#1-simplicity-first)
-2. [The API is the most important part of quality](#2-the-api-is-the-most-important-part-of-quality)
-3. [Design before implementation](#3-design-before-implementation)
-4. [Reuse, DRY, and separation of concerns](#4-reuse-dry-and-separation-of-concerns)
-5. [Declarative over imperative](#5-declarative-over-imperative)
-6. [Never precompute a size to predict whether something fits](#6-never-precompute-a-size-to-predict-whether-something-fits)
-7. [CLI parameters over environment variables](#7-cli-parameters-over-environment-variables)
-8. [Embedded DSLs should reuse host-language syntax](#8-embedded-dsls-should-reuse-host-language-syntax)
+2. [Maximize signal-to-noise](#2-maximize-signal-to-noise)
+3. [The API is the most important part of quality](#3-the-api-is-the-most-important-part-of-quality)
```
[the remark this answers is not in the document]
Comment (sergey-shandar) [reply DESIGN.md]: Fixed by preserving the existing section numbers and anchors. `Maximize signal-to-noise` is now §9, so existing references to API quality (§2), size checks (§6), and embedded DSLs (§8) remain valid without touching unrelated files.

**Review by Trinidadec — APPROVED** [review]
